diff --git a/challenge/dns01/dns_challenge.go b/challenge/dns01/dns_challenge.go
index ca24e7af..54e4545e 100644
--- a/challenge/dns01/dns_challenge.go
+++ b/challenge/dns01/dns_challenge.go
@@ -131,7 +131,7 @@ func (c *Challenge) Solve(authz acme.Authorization) error {
 	err = wait.For("propagation", timeout, interval, func() (bool, error) {
 		stop, errP := c.preCheck.call(domain, fqdn, value)
 		if !stop || errP != nil {
-			log.Infof("[%s] acme: Waiting for DNS record propagation.", domain)
+			log.Infof("[%s] acme: Waiting for DNS record propagation: %s", domain, errP)
 		}
 		return stop, errP
 	})
@@ -196,7 +196,10 @@ func getChallengeFqdn(domain string) string {
 
 		// Check if the domain has CNAME then use that
 		if err == nil && r.Rcode == dns.RcodeSuccess {
-			fqdn = updateDomainWithCName(r, fqdn)
+			updated := updateDomainWithCName(r, fqdn)
+		        log.Infof("[%s] Found CNAME for requested domain: %s -> %s", domain, fqdn, updated)
+
+			fqdn = updated
 			continue
 		}
 
